docs/website/contents/for-developers/HardWonWisdom.md

```diff
@@ -348,3 +348,162 @@ The key corresponding parts of the implementation are as follows.
+Note that the second ledger state in that specification necessarily arises from a block in the era, since only the first ledger state in an era does not arise from a block in that era.
+Let `y` be that block; the exact safe zone constraint is `slot(y) + d < firstSlot(e)`.
+(Lastly, recall once more that the ledger state sequence in question is intentionally excluding any arising from EBBs.)
+
+That interpretation of the safe zone constraint explicitly only considers the evolution of `singleEraTransition` _within a single era_.
+In particular, it did not at all constrain the output of `singleEraTransition` for the initial ledger state of the era.
+The rest of this subsection discusses that, including notably that that initial output is always `Nothing` in the fundamental use case, in which the end of an era cannot be known before any of its block exist.
+
+In the fundamental motivating use case of the HFC, `singleEraTransition` will return `Nothing` for the first ledger state of each era, and subsequent governance events within that era on the chain will cause it to return `Just` for the remaining suffix of that chain's ledger states in this era.
+However, there are a total of three possible trajectories of the `singleEraTransition` within an era on some chain.
+
+- EXTENSIBLE trajectory.
+  It starts with `Nothing` and ends with `Just`, as in the fundamental motivating use case.
+  Note that once it becomes `Just` it will always remain `Just` (with the same epoch number `e`) for the rest of this chain, by the monotonicity of `singleEraTransition` within an era of some chain.
+  This is the only case in which the above single-era interpretation of the safe zone applies, because it's the only case that involves both `Nothing` and `Just` in the same era.
+- NEVER trajectory.
+  It starts a `Nothing` and will necessarily remain `Nothing` on the entire chain.
+  For example, `singleEraTransition _cfg _params _start _st = Nothing` is a compatible definition.
+  In this case, any safe zone value would satisfy the high-level rule, because the era never ends---the implementation names this an _indefinite_ safe zone.
+- IMMEDIATE trajectory.
+  It starts as `Just e`.
+  Note that it will always remain `Just e` for the rest of this chain, by the monotonicity of `singleEraTransition` within an era of some chain.
+  The IMMEDIATE trajectory can arise in two ways.
+    - Case IMMEDIATE-INDEPENDENT.
+      It's possible that the configuration data determines the end of the era, entirely independent of the preceding chain.
+      For example, `singleEraTransition cfg _params _start _st = Just $ foo cfg + 42` is a valid definition.
+      In this case, any safe zone value would satisfy the high-level rule, because the end of the era is determined even before the start is.
+      TODO but `singleEraTransition` cannot be applied without the translate state, so what is the current code doing in this case?
```

Comment:
I explored this TODO with the `StaticTransition` type in PR https://github.com/input-output-hk/ouroboros-consensus/pull/364

Reply:
I am not exactly sure what "current code" you are referring to. The HFC cross-era ticking logic can just use the translated state, so you probably don't mean that.
Hence, you probably mean forecasting/`Summary` construction etc. as these currently do not translate ledger states, so they use `TransitionUnkown` and hence construct the upper bound of a IMMEDIATE-INDEPENDENT era (before having ticked into it) only based on the safe zone, even though the end is already known statically.

Reply:
That's 100% what I meant, yep.

Arguably, this is a bug in the HFC, currently. It contains logic that assumes future eras will contain at least one safe zone's worth of slots, but that's not necessarily true if those eras are configured via `TriggerHardForkAtEpoch` (which induces the IMMEDIATE-INDEPENDENT trajectory).

Reply:
I rephrased the TODO text to be much much more precise